components/ml/org.wso2.carbon.ml.rest.api/src/main/java/org/wso2/carbon/ml/rest/api/ModelApiV10.java

```diff
@@ -187,6 +184,45 @@ public Response predict(@Multipart("modelId") long modelId, @Multipart("dataForm
         }
     }
 
+    /**
+     * Predict using a file and return predictions as a CSV.
+     */
+    @POST
+    @Path("/predictionStreams")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response streamingPredict(@Multipart("modelId") long modelId, @Multipart("dataFormat") String dataFormat,
+                            @Multipart("file") InputStream inputStream) {
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        int tenantId = carbonContext.getTenantId();
+        String userName = carbonContext.getUsername();
+        try {
+            // validate input parameters
+            // if it is a file upload, check whether the file is sent
+            if (inputStream == null || inputStream.available() == 0) {
+                logger.error("Cannot read the file.");
```

Comment:
define a String msg and assign the error message into it. Also in error message please add for which model id etc.